bciers/apps/reporting/src/app/components/taskList/5_signOffSubmit.ts

```diff
@@ -1,47 +1,64 @@
 import { TaskListElement } from "@bciers/components/navigation/reportingTaskList/types";
+import { getReportNeedsVerification } from "@reporting/src/app/utils/getReportNeedsVerification";
 
 export enum ActivePage {
   "Verification" = 0,
   "Attachments",
   "FinalReview",
   "SignOff",
 }
-
 export const getSignOffAndSubmitSteps: (
   versionId: number,
   activeIndex?: ActivePage | number,
-) => TaskListElement[] = (versionId, activeIndex = undefined) => {
+) => Promise<TaskListElement[]> = async (
+  versionId,
+  activeIndex = undefined,
+) => {
+  // Fetch needsVerification within the function
+  const needsVerification = await getReportNeedsVerification(versionId);
+  console.log(needsVerification);
+  // Build the TaskListElement array
+  const elements: TaskListElement[] = [
+    {
+      type: "Page", // Set the type to "Page"
+      title: "Verification",
+      link: `/reports/${versionId}/verification`,
+      isActive: activeIndex === ActivePage.Verification,
+    },
+    {
+      type: "Page", // Set the type to "Page"
+      title: "Attachments",
+      link: `/reports/${versionId}/attachments`,
+      isActive: activeIndex === ActivePage.Attachments,
+    },
+    {
+      type: "Page", // Set the type to "Page"
+      title: "Final review",
+      link: `/reports/${versionId}/final-review`,
+      isActive: activeIndex === ActivePage.FinalReview,
+    },
+    {
+      type: "Page", // Set the type to "Page"
+      title: "Sign-off",
+      link: `/reports/${versionId}/sign-off`,
+      isActive: activeIndex === ActivePage.SignOff,
+    },
+  ];
+  // Conditionally filter out "Verification" and "Attachments" based on needsVerification is false
+  const filteredElements: TaskListElement[] = elements.filter((element) => {
+    if (!needsVerification) {
+      return (
+        element.title !== "Verification" && element.title !== "Attachments"
```

Comment:
The Attachments page contains more than just the verification statement (confidentiality request, extra WCI info, etc.) so we need to keep it there no matter what